### PR TITLE
Apply StyleCop to project

### PR DESCRIPTION
**Pull request by justing-bq**
Apply StyleCop to project

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by kylepbit — APPROVED** [review]

File: Amazon.Ion.ObjectMapper/IonSerializer.cs [hunk 1]
```diff
@@ -1,51 +1,66 @@
...
     {
         BINARY,
-        TEXT, 
-        PRETTY_TEXT
+        TEXT,
+        PRETTY_TEXT,
     }
 
-    public interface IonReaderFactory
+    public interface IIonReaderFactory
```
[the remark this answers is not in the document]
Comment (justing-bq) [reply Amazon.Ion.ObjectMapper/IonSerializer.cs:43]: Pardon me, where is the structural change? This code is just renaming the reader factory.

[the remark this answers is not in the document]
Comment (kylepbit) [reply Amazon.Ion.ObjectMapper/IonSerializer.cs:43]: Sorry - the renaming of IonReaderFactor to IIonReaderFactory - I think StyleCop says interfaces should be prefixed with `I` which is a "structural" change in that it changes class names. Worth calling out explicitly.

File: Amazon.Ion.ObjectMapper.Test/TestObjects.cs [hunk 2]
```diff
@@ -244,26 +262,26 @@ public School()
...
         }
     }
 
     public class Teacher
     {
-        public readonly string firstName;
-        public readonly string lastName;
-        public string department;
-        public readonly DateTime? birthDate;
+        internal readonly string firstName;
```
Comment (alpian) [inline Amazon.Ion.ObjectMapper.Test/TestObjects.cs:281]: StyleCop doesn't realise that these are public for a good reason.

**Review by alpian — CHANGES_REQUESTED** [review]
Not convinced StyleCop is needed. If it is needed it needs to be customized for the deliberate "violations" we need. e.g. public fields that can be internal but are deliberately not.

**Comment by alpian** [comment]
Going to close this one, it will be completely different when we actually make this change.

**Pull request closed without merging**